blob-storage-staging.tf

```diff
@@ -0,0 +1,80 @@
+locals {
+  stripped_product_stg  = "${replace(var.product, "-", "")}"
+  account_name_stg      = "${local.stripped_product_stg}${var.env}staging"
+  mgmt_network_name_stg = "core-cftptl-intsvc-vnet"
+  mgmt_network_rg_name_stg = "aks-infra-cftptl-intsvc-rg"
+  prod_hostname_stg     = "${local.stripped_product_stg}stg.${var.external_hostname}"
+  nonprod_hostname_stg  = "${local.stripped_product_stg}.${var.env}stg.${var.external_hostname}"
+  external_hostname_stg = "${ var.env == "prod" ? local.prod_hostname_stg : local.nonprod_hostname_stg}"
+
+  // for each client service two containers are created: one named after the service
+  // and another one, named {service_name}-rejected, for storing envelopes rejected by process
+  client_containers_stg = ["bulkscan", "cmc", "crime", "divorce", "finrem", "probate", "sscs", "publiclaw"]
+}
+
+data "azurerm_subnet" "trusted_subnet_stg" {
+  name                 = "${local.trusted_vnet_subnet_name_stg}"
+  virtual_network_name = "${local.trusted_vnet_name_stg}"
+  resource_group_name  = "${local.trusted_vnet_resource_group_stg}"
+}
+
+data "azurerm_subnet" "jenkins_subnet_stg" {
+  provider             = "azurerm.mgmt"
+  name                 = "iaas"
+  virtual_network_name = "${local.mgmt_network_name_stg}"
+  resource_group_name  = "${local.mgmt_network_rg_name_stg}"
+}
+
+resource "azurerm_storage_account" "storage_account_staging" {
+  name                = "${local.account_name_stg}"
+  resource_group_name = "${azurerm_resource_group.rg.name}"
+
+  location                 = "${azurerm_resource_group.rg.location}"
+  account_tier             = "Standard"
+  account_replication_type = "LRS"
+  account_kind             = "BlobStorage"
+
+  custom_domain {
+    name          = "${local.external_hostname_stg}"
+    use_subdomain = "false"
+  }
+
+  network_rules {
+    virtual_network_subnet_ids = ["${data.azurerm_subnet.jenkins_subnet_stg.id}", "${data.azurerm_subnet.jenkins_subnet_stg.id}"]
```

Comment:
there are 2 identical ids here - is this correct?

Reply:
good catch one needs to be Jenkins and other should be Jenkins subnet. Will update